### PR TITLE
TileData: Add Typescript Types

### PR DESCRIPTION
**Pull request by rlingineni**
- Adding Typescript Types 
- Remove Controls from Don't examples
image

### CODE REVIEW

File: packages/gestalt/dist/index.d.ts [hunk 1]
```diff
@@ -1920,6 +1939,28 @@ interface TextFieldProps {
   value?: string | undefined;
 }
 
+interface TileDataProps {
+   color?: DataVisualizationColors,
+   disabled?: boolean,
+   id?: string,
+   onTap?:(
```
Comment (AlbertCarreras) [inline packages/gestalt/dist/index.d.ts:1946]: Can you use AbstractEventHandler here?

```
AbstractEventHandler<
  | React.MouseEvent<HTMLDivElement>
  | React.KeyboardEvent<HTMLDivElement>
  | React.MouseEvent<HTMLAnchorElement>
  | React.KeyboardEvent<HTMLAnchorElement>,
  { selected: boolean, 
     id?: string 
    }
>

```


Also, this looks incorrect:
`React.MouseEvent<HTMLDivElement, MouseEvent`

**Review by AlbertCarreras — APPROVED** [review]
R+ but review comment

File: packages/gestalt/dist/index.d.ts (hunk 1 above)
Comment (rlingineni) [reply packages/gestalt/dist/index.d.ts:1946]: Okay sure

Comment (rlingineni) [reply packages/gestalt/dist/index.d.ts]: Fixed, but is there any way to validate there's no syntax error or something?

**Review by AlbertCarreras — APPROVED** [review]

**Pull request merged**